### DISCUSSION TITLE
Message passing aggregation arbitrary function

### DISCUSSION THREAD

**Discussion by diegocervera** [pyg-team/pytorch_geometric#3510] [Q&A]
Currently the MessagePassing class has three different aggregation schemes: 'add', 'mean' or 'max'. Would it be possible to use an arbitrary function to aggregate the messages, e.g. a GRU?

**Comment by wsad1** [comment]
You could override the `aggregate` function as done in [PNAConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.PNAConv) or [GENConv](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html#torch_geometric.nn.conv.GENConv).

**Comment by rusty1s** [comment]
Useful relevant discussion: https://github.com/pyg-team/pytorch_geometric/discussions/3197
